src/libraries/Microsoft.Extensions.Logging/gen/LoggerMessageGenerator.Emitter.cs

```diff
@@ -0,0 +1,589 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Threading;
+
+namespace Microsoft.Extensions.Logging.Generators
+{
+    public partial class LoggerMessageGenerator
+    {
+        internal class Emitter
+        {
+            // The maximum arity of LoggerMessage.Define.
+            private const int MaxLoggerMessageDefineArguments = 6;
+
+            private readonly string _generatedCodeAttribute =
+                $"global::System.CodeDom.Compiler.GeneratedCodeAttribute(" +
+                $"\"{typeof(Emitter).Assembly.GetName().Name}\", " +
+                $"\"{typeof(Emitter).Assembly.GetName().Version}\")";
+            private readonly Stack<StringBuilder> _builders = new ();
```

Comment:
>Stack [](start = 29, length = 5)

I assume there is no thread safty issue here. I mean we cannot have multiple threads accessing the stack in the same time.

Reply:
I don't believe there a thread safety issue here - this is an instance field. And we create a new Emitter object each time `Execute` is called on the source generator.

That being said, this cache will just go away after https://github.com/dotnet/runtime/pull/51064#discussion_r612506811 is addressed.